apps/supernova/src/lib/createSilencesSlice.tsx

```diff
@@ -89,142 +124,139 @@ const validateTemplates = (templates: any) => {
 }
 
-// @ts-expect-error TS(7006) FIXME: Parameter 'set' implicitly has an 'any' type.
-const createSilencesSlice = (set, get, options) => ({
-  silences: {
-    ...initialSilencesState,
-
-    // silence templates for maintanance
-    templates: options?.silenceTemplates ? validateTemplates(options?.silenceTemplates) : [],
-    excludedLabels: validateExcludedLabels(options?.silenceExcludedLabels),
-    actions: {
-      setSilencesStatus: (status: any) =>
-        set(
-          (state: any) => ({
-            silences: { ...state.silences, status: status },
-          }),
-          false,
-          "silences.setSilencesStatus"
-        ),
-      setSilencesRegEx: (regEx: any) =>
-        set(
-          (state: any) => ({
-            silences: { ...state.silences, regEx: regEx },
-          }),
-          false,
-          "silences.setSilencesRegEx"
-        ),
-
-      getSilenceById: (id: any) => {
-        return get().silences.items.find((silence: any) => silence.id === id)
-      },
+const createSilencesSlice: (options?: Record<string, any>) => StateCreator<AppState, [], [], SilencesSlice> =
+  (options) => (set, get, store) => ({
+    silences: {
+      ...initialSilencesState,
 
-      setSilences: ({ items }: any) => {
-        if (!items) return
-
-        set((state: any) => ({
-          silences: {
-            ...state.silences,
-            items: items,
-            updatedAt: Date.now(),
-          },
-        })),
-          false,
-          "silences.setSilencesData"
-      },
+      // silence templates for maintanance
+      templates: options?.silenceTemplates ? validateTemplates(options?.silenceTemplates) : [],
+      excludedLabels: validateExcludedLabels(options?.silenceExcludedLabels),
+      actions: {
+        setSilencesStatus: (status) =>
+          set(
+            (state: any) => ({
+              silences: { ...state.silences, status: status },
+            }),
+            false
+          ),
+        setSilencesRegEx: (regEx) =>
+          set(
+            (state: any) => ({
+              silences: { ...state.silences, regEx: regEx },
+            }),
+            false
+          ),
+
+        getSilenceById: (id) => {
+          return get().silences.items.find((silence: any) => silence.id === id)
+        },
+
+        setSilences: ({ items }) => {
+          if (!items) return
+
+          set((state: any) => ({
+            silences: {
+              ...state.silences,
+              items: items,
+              updatedAt: Date.now(),
+            },
+          })),
+            false
+        },
 
-      /* 
+        /* 
       Given an alert fingerprint, this function returns all silences referenced by silencingBy. 
       */
-      getMappingSilences: (alert: any) => {
-        if (!alert) return
-        const externalSilences = get().silences.items
-        let silencedBy = alert?.status?.silencedBy || []
+        getMappingSilences: (alert) => {
+          if (!alert) return []
```

Comment:
Have you verified that changing the return to empty array doesn't break anything? If the place where the mapping silences are consumed doesn't explicitly check for empty array this might cause strange behaviour